modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java

```diff
@@ -4190,4 +4219,55 @@ public boolean addTimeoutHandler() {
             "mappings", mappings,
             "super", super.toString());
     }
+
+    /** {@inheritDoc} */
+    @Override public void savepoint(String name, boolean overwrite) throws IgniteCheckedException {
+        synchronized (this) {
+            checkValid();
+
+            if (state() != ACTIVE) {
+                throw new IllegalStateException("Trying to create savepoint not in " + ACTIVE + " transaction." +
+                    " Savepoint will not be created.");
+            }
+
+            ((IgniteTxStateImpl)txState).savepoint(name, overwrite);
```

Comment:
Can we add `savepoint`, `releaseSavepoint`, `rollbackToSavepoint` methods to an interface?

Reply:
Yes, but only 1 class implementats them. Do we really need redundant methods?